### PR TITLE
Email verification required as a Mixins

### PR DESCRIPTION
**Pull request by simo97**
feat(account): Added a new mixins to  enable email verification control on a class-based views.

### CODE REVIEW

**Comment by simo97** [comment]
Oh ok will change that quickly.

File: allauth/account/mixins.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""
+This module contain a class mixin design to bring email verification
+to class based views in django-allauth
+It is used like django's LoginRequiedMixin, so when using this
```
Comment (pennersr) [inline allauth/account/mixins.py:4]:
```suggestion
It is used like django's LoginRequiredMixin, so when using this
```

File: allauth/account/mixins.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+there is no need to use LoginRequiedMixin fro django
```
Comment (pennersr) [inline allauth/account/mixins.py:5]:
```suggestion
there is no need to use LoginRequiredMixin fro django
```

File: docs/decorators.rst [hunk 3]
```diff
@@ -24,3 +24,20 @@ The behavior is as follows:
...
+
+Mixins
+======
+
+Verified E-mail Required
+------------------------
+
+In the case you are using a class-based view, there is a mixin available
+in ``allauth.account.mixins.VerifiedEmailRequiredMixin`` which is use in 
+any class-based view from django, you can use it as follow ::
```
Comment (pennersr) [inline docs/decorators.rst:36]:
```suggestion
any class-based view from django, you can use it as follows ::
```

File: docs/decorators.rst [hunk 4]
```diff
@@ -24,3 +24,20 @@ The behavior is as follows:
...
+
+
+    from django.views.generics import ListView
+    from allauth.account.mixins import VerifiedEmailRequiredMixin
+
+    class YouClassName(VerifiedEmailRequiredMixin, ListView):
```
Comment (pennersr) [inline docs/decorators.rst:42]:
```suggestion
    class YourClassName(VerifiedEmailRequiredMixin, ListView):
```

**Review by pennersr — CHANGES_REQUESTED** [review]

File: allauth/account/mixins.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+"""
+This module contain a class mixin design to bring email verification
+to class based views in django-allauth
+It is used like django's LoginRequiredMixin, so when using this
+there is no need to use LoginRequiredMixin fro django
```
Comment (pennersr) [inline allauth/account/mixins.py:5]:
```suggestion
there is no need to use LoginRequiredMixin from Django
```

File: allauth/account/mixins.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
+"""
+This module contain a class mixin design to bring email verification
+to class based views in django-allauth
+It is used like django's LoginRequiredMixin, so when using this
```
Comment (pennersr) [inline allauth/account/mixins.py:4]:
```suggestion
It is used like Django's LoginRequiredMixin, so when using this
```

**Review by pennersr — CHANGES_REQUESTED** [review]

**Pull request closed without merging**